### PR TITLE
[Merged by Bors] - feat: add max_bytes to connector consumer option

### LINKED ISSUE #3210: add max_bytes to connector consumer option

### PR DESCRIPTION
**Pull request by morenol**
Closes #3210

### CODE REVIEW

**Review by nacardin — COMMENTED** [review]
Wouldn't it be cleaner to implement a custom deserializer/serialzier for `ByteSize` instead of a separate string field?

You could mark the field with `#[serde(with = "module")] ` https://serde.rs/field-attrs.html#with

**Comment by morenol** [comment]
> Wouldn't it be cleaner to implement a custom deserializer/serialzier for `ByteSize` instead of a separate string field?
> 
> You could mark the field with `#[serde(with = "module")] ` https://serde.rs/field-attrs.html#with

addressed in https://github.com/infinyon/fluvio/pull/3209/commits/692b9ff26e6b25fdafe430a41829389608e2f874

Now field is `max_bytes`. I don't like the inconsistency between `max_bytes` and `batch-size`. Maybe I should update it too for `batch_size`?

**Comment by morenol** [comment]
> > Wouldn't it be cleaner to implement a custom deserializer/serialzier for `ByteSize` instead of a separate string field?
> > You could mark the field with `#[serde(with = "module")] ` https://serde.rs/field-attrs.html#with
> 
> addressed in [692b9ff](https://github.com/infinyon/fluvio/commit/692b9ff26e6b25fdafe430a41829389608e2f874)
> 
> Now field is `max_bytes`. I don't like the inconsistency between `max_bytes` and `batch-size`. Maybe I should update it too for `batch_size`?

Addressed my concern on https://github.com/infinyon/fluvio/pull/3209/commits/b7bc0e233daf07c061f6a19744a55c98063b323e

**Review by nacardin — APPROVED** [review]
Nice!

File: crates/fluvio-connector-package/test-data/connectors/full-config.yaml [hunk 1]
```diff
@@ -11,6 +11,7 @@ meta:
     compression: gzip
   consumer:
     partition: 10
+    max_bytes: "1 MB"
   secrets:
```
Comment (sehz) [inline crates/fluvio-connector-package/test-data/connectors/full-config.yaml:15]: Can we add test cases for just regular numbers? For example, this should work
```
consumer:
      max_bytes: 13933
```
```

Comment (morenol) [reply crates/fluvio-connector-package/test-data/connectors/full-config.yaml:15]: Ok, I will update then to use:  https://docs.rs/bytesize-serde/latest/bytesize_serde/ that seems to handle that case

Comment (morenol) [reply crates/fluvio-connector-package/test-data/connectors/full-config.yaml:15]: Addressed in https://github.com/infinyon/fluvio/pull/3209/commits/869d4ef8b0b8dac2543311fdde36a80b359c2aec (also added test there)

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**